src/bootstrap/compile.rs

```diff
@@ -544,7 +544,7 @@ pub fn rustc_cargo_env(builder: &Builder, cargo: &mut Command) {
          .env("CFG_PREFIX", builder.config.prefix.clone().unwrap_or_default())
          .env("CFG_CODEGEN_BACKENDS_DIR", &builder.config.rust_codegen_backends_dir);
 
-    let libdir_relative = builder.config.libdir_relative().unwrap_or(Path::new("lib"));
+    let libdir_relative = builder.config.libdir_relative().unwrap_or_else(|| Path::new("lib"));
```

Comment:
Hm, ideally, using the `_else` variant could be a great indicator whether the computation is actually expensive. This change, for example, is not necessary: It [is](https://doc.rust-lang.org/1.29.2/src/std/path.rs.html#1698-1700) just a cast.

I'm fine with changing all function calls to `_else` for now, though.

Reply:
Yeah most of these `_else` are line noise with no positive impact. Many wrap functions like `String::new`, `Vec::new`, tuple struct constructors, etc. that are literally free.

Reply:
@rkruppe well, most are `to_string()`, which are not free.

Reply:
Fair, I haven't actually counted, but I did see a significant number of them.

Reply:
@rkruppe you are 100% right about `Vec/String::new`, though; I knew they didn't allocate, but I didn't notice they had become `const fn`.